pkg/apis/policy/v1alpha1/propagation_types.go

```diff
@@ -660,6 +687,59 @@ const (
 	LazyActivation ActivationPreference = "Lazy"
 )
 
+// SchedulePriority defines how Karmada should resolve the priority and preemption policy
+// for workload scheduling.
+type SchedulePriority struct {
+	// PriorityClassSource specifies where Karmada should look for the PriorityClass definition.
+	// Available options:
+	// - KubePriorityClass (default): Uses Kubernetes PriorityClass (scheduling.k8s.io/v1)
+	// - PodPriorityClass: Uses PriorityClassName from PodTemplate (PodSpec.PriorityClassName)
+	// - FederatedPriorityClass: Uses Karmada FederatedPriorityClass (not yet implemented)
+	//
+	// +kubebuilder:default="KubePriorityClass"
+	// +kubebuilder:validation:Enum=FederatedPriorityClass;KubePriorityClass;PodPriorityClass
+	// +optional
+	PriorityClassSource PriorityClassSource `json:"priorityClassSource,omitempty"`
+
+	// PriorityClassName specifies which PriorityClass to use. Its behavior depends on PriorityClassSource:
+	//
+	// PriorityClassName behavior:
+	//
+	// For KubePriorityClass:
+	// - If specified: Uses the named Kubernetes PriorityClass
+	// - If empty: Uses cluster's global default PriorityClass
+	// - If neither exists: Sets priority=0 and preemptionPolicy=Never
+	//
+	// For PodPriorityClass:
+	// - Uses PriorityClassName from PodTemplate
+	// - Falls back to global default if not found
+	// - If no valid PriorityClass found: Sets priority=0 and preemptionPolicy=Never
+	//
+	// For FederatedPriorityClass:
+	// - Reserved for future use
+	// - Current behavior undefined
+	//
+	// +optional
```

Comment:
```suggestion
	// +required
```
Can we restrict this as a required in the current version, just hope to work out a simple version quickly.

@seanlaii In your use cases, will you set a specific PriorityClassName for each workload, or will you rely on the default one?

Reply:
Sure, we currently set a default priority class for every workload if no priority class is set.

Reply:
Changed it to required.

Reply:
> Sure, we currently set a default priority class for every workload if no priority class is set.

+1 
I think that making the configuration clearer, and also a relief for the implementation.